mapswipe_workers/mapswipe_workers/generate_stats/generate_stats.py

```diff
@@ -0,0 +1,46 @@
+from mapswipe_workers.definitions import logger
+from mapswipe_workers.definitions import DATA_PATH
+
+from mapswipe_workers.generate_stats import project_stats
+from mapswipe_workers.generate_stats import overall_stats
+
+
+def generate_stats(project_id_list, all_projects=False):
+
+    projects_info_filename = f'{DATA_PATH}/api-data/projects/projects_static.csv'
+    projects_df = overall_stats.get_project_static_info(projects_info_filename)
+
+    projects_info_dynamic_filename = f'{DATA_PATH}/api-data/projects/projects_dynamic.csv'
+    projects_dynamic_df = overall_stats.load_project_info_dynamic(projects_info_dynamic_filename)
+
+    if all_projects:
+        project_id_list = projects_df['project_id'].to_list()
+        logger.info(f'will generate stats for all projects.')
+
+    logger.info(f'will generate stats for: {project_id_list}')
+
+    # get per project stats and aggregate based on task_id
+    for project_id in project_id_list:
+
+        # check if project id is correct
+        if not project_id in projects_df['project_id'].to_list():
```

Comment:
It would probably make sense to factor the `projects_df['project_id'].to_list()` out of the for loop, so you don't run it `len(project_id_list)` times, as it's probably quite expensive.